simplepso/pso.py

```diff
@@ -348,6 +325,30 @@ def run(self, num_particles, num_iterations, num_processes=1,
         self.values = np.array(values[:g])
         self.history = np.array(history[:g, :])
 
+    def _calc_fitness_from_array(self, traj, num_sim):
+        index_names = traj.index.names
+        traj.reset_index(inplace=True)
+        for i, part in enumerate(self.population):
+            start = i * num_sim
+            end = (i + 1) * num_sim
+            tmp_results = traj.loc[
+                traj.simulation.isin(list(range(start, end)))]
+            # tmp_results = tmp_results.T.unstack('simulation')
+            tmp_results.set_index(index_names, inplace=True)
+            error = self.cost_function(tmp_results)
```

Comment:
The `cost_function` function for ssa is different from the one for odes. I think it would be nice to explain the differences in the documentation.

Reply:
Added. I am still debating the best way to add examples for each. Inline documentation would be ideal, but I don't see a clean way of doing it with defining so many functions. Jupyter notebooks would be nice too, but multi-processing with notebooks doesn't work on windows.

Reply:
Wouldn't it be more clear to remove the cost_function from the class init and add it in the run function? That way you would have a run_ode and a run_ssa with well-defined arguments?

Reply:
Good point. I moved it around. I didn't rename run_ode since it can be to run other cost functions, including other SSA implementations. For instance, StochKitSimulators or BngSimulator, which handle SSA simulation differently. You also don't need pysb for running PSO either. WIth the serial executor you need to install it, but the tests optimize random functions.